### PR TITLE
Added log rotation feature and console/file logging config options

### PR DESCRIPTION
**Pull request by Shoukozumi**
- Added log rotation feature, currently set to rotate for every 12h or if log file size exceeds 2 Gb. If the log directory exceeds 50 Gb, old log files will be deleted.
- Added config options for toggling console and file logging.
- Changed config options for log file storage, now writing log files to a directory instead of a single file.
- Updated config to allow log rotation size, log rotation interval, and log directory max size specification.

### CODE REVIEW

**Comment by legleux** [comment]
Don't merge this just yet.

File: example-config.json
Comment (natenichols) [inline example-config.json]: How hard would it be to be able to specify that in GB

Like
```json
"log_rotation_size": "2GB" // or "2MB"
```

If it's hard, don't worry about it, but it would be way more readable.

File: src/main.cpp
Comment (natenichols) [inline src/main.cpp]: nit: just say 

```cpp
&& config.at("log_to_console").as_bool()
```

File: src/main.cpp
Comment (natenichols) [inline src/main.cpp]: ditto
```cpp
config.at("log_to_file").as_bool()
```

File: src/main.cpp
Comment (natenichols) [inline src/main.cpp]: I would add a check here that is 

```cpp
if (!config.at("log_directory").is_string())
    throw std::runtime_error("log directory must be a string");
```

or something similar. that will be more helpful than the error boost.json throws, which is 
`bad_cast` or something.

Comment (natenichols) [reply src/main.cpp]: Our config parsing happens all over the place, which is bad. We should be parsing the config on startup and throwing errors immediately if its wrong. 

#151 fixes that, tho we have to get reviews on it before we can merge...

File: src/main.cpp
Comment (natenichols) [inline src/main.cpp]: So, does rotation happen when the `file > dirSize` or every `rotationPeriod` seconds, or both?

I.E. if we hit our limit, does that cancel out the time based rotation, or will we see something like this:
```
debug001.log -> 2GB
debug002.log -> 2GB
debug003.log -> 1GB // here we hit the time based rotation when the file is only 1GB
debug004.log -> 2GB
```

**Review by natenichols — CHANGES_REQUESTED** [review]
Awesome work on this! 

This change make log rotation highly configurable, which will be super useful.

I have a couple of questions about how functionality behaves, and I can't actually test it out right now since I'm not on my workstation. I will test it out locally next tuesday.

File: src/main.cpp
Comment (Shoukozumi) [reply src/main.cpp]: The rotation happens whenever either of the two conditions is met (when the main log file exceeds the limit or enough time passes since the last rotation). So yes, when the log is rotated because of the file size limit, the time will reset.

File: src/main.cpp
Comment (Shoukozumi) [reply src/main.cpp:113]: Fixed

File: src/main.cpp
Comment (Shoukozumi) [reply src/main.cpp]: Fixed

File: README.md
Comment (natenichols) [inline README.md]: `rippled nodes`

File: README.md
Comment (natenichols) [inline README.md:148]: The config documentation section looks fantastic. Nice work

**Review by natenichols — APPROVED** [review]
I'll run your changes locally tomorrow, so don't merge until then (make sure @legleux is good to merge this too) but this looks good to me.

File: README.md
Comment (Shoukozumi) [reply README.md]: Fixed

File: src/main.cpp
Comment (cjcobb23) [inline src/main.cpp]: I think we should default to logging to console

Comment (Shoukozumi) [reply src/main.cpp]: Fixed

**Comment by natenichols** [comment]
Tested this locally, seems to work as expected.

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]: Why did the `backend` README.md get added to this PR?

**Comment by legleux** [comment]
LGTM! :shipit:

**Pull request merged**
